apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/CompletionScreen.tsx

```diff
@@ -1,28 +1,25 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useContext } from "react";
 import { Flex } from "@ledgerhq/react-ui";
-import { useHistory } from "react-router-dom";
 import { useDispatch, useSelector } from "react-redux";
 
+import { useStartPostOnboardingCallback } from "@ledgerhq/live-common/postOnboarding/hooks/index";
+import ConnectNano from "../../Onboarding/Screens/Tutorial/assets/connectNano.png";
+import Image from "~/renderer/components/Image";
 import { saveSettings } from "~/renderer/actions/settings";
-import DeviceIllustration from "~/renderer/components/DeviceIllustration";
 import { getCurrentDevice } from "~/renderer/reducers/devices";
 
 const GO_TO_POSTONBOARDING_TIMEOUT = 5000;
 
 const CompletionScreen = () => {
-  const history = useHistory();
   const dispatch = useDispatch();
   const device = useSelector(getCurrentDevice);
 
+  const handleInitPostOnboarding = useStartPostOnboardingCallback(device?.modelId, true);
+
   useEffect(() => {
     dispatch(saveSettings({ hasCompletedOnboarding: true }));
-    // TODO: replace with this one when post onboarding is ready
-    // setTimeout(
-    //   () => history.push("/onboarding/sync/post-onboarding"),
-    //   GO_TO_POSTONBOARDING_TIMEOUT,
-    // );
-    setTimeout(() => history.push("/"), GO_TO_POSTONBOARDING_TIMEOUT);
-  }, [history, dispatch]);
+    setTimeout(() => handleInitPostOnboarding(), GO_TO_POSTONBOARDING_TIMEOUT);
```

Comment:
Whenever you set a timeout or do something asynchronous in a `useEffect` you need to return a cleanup function that will do some cleanup for instance in case the component is unmounted, in order to avoid memory leaks.

Here you can do:
```
useEffect(() => {
    dispatch(saveSettings({ hasCompletedOnboarding: true }));
    const timeout = setTimeout(handleInitPostOnboarding, GO_TO_POST_ONBOARDING_TIMEOUT);
    return () => { // cleanup function 
      clearTimeout(timeout);
    }
}, [dispatch, handleInitPostOnboarding]);
```

cf. [doc on effects with cleanup](https://en.reactjs.org/docs/hooks-effect.html#effects-with-cleanup) very important React concept so please read this.